evm_arithmetization/src/prover.rs

```diff
@@ -376,7 +569,51 @@ pub mod testing {
         let initial_stack = vec![];
         let initial_offset = KERNEL.global_labels["main"];
         let mut interpreter: Interpreter<F> =
-            Interpreter::new_with_generation_inputs(initial_offset, initial_stack, inputs);
-        interpreter.run()
+            Interpreter::new_with_generation_inputs(initial_offset, initial_stack, &inputs, None);
+        interpreter.run()?;
+        Ok(())
+    }
+
+    pub fn prove_all_segments<F, C, const D: usize>(
+        all_stark: &AllStark<F, D>,
+        config: &StarkConfig,
+        inputs: GenerationInputs,
+        max_cpu_len: usize,
+        timing: &mut TimingTree,
+        abort_signal: Option<Arc<AtomicBool>>,
+    ) -> Result<Vec<AllProof<F, C, D>>>
+    where
+        F: RichField + Extendable<D>,
+        C: GenericConfig<D, F = F>,
+    {
+        let mut segment_idx = 0;
+        let mut proofs = vec![];
+        while let Some(proof) = prove(
+            all_stark,
+            config,
+            inputs.clone(),
+            max_cpu_len,
+            segment_idx,
+            timing,
+            abort_signal.clone(),
+        )? {
+            segment_idx += 1;
+            proofs.push(proof);
+        }
+        Ok(proofs)
+    }
+
+    pub fn prove_all_segments_interpreter<F>(
```

Comment:
Should probably be called `simulate` similarly to the one above, as we don't prove anything here.